src/components/SecretsModal/Annotations.js

```diff
@@ -0,0 +1,82 @@
+/*
+Copyright 2019 The Tekton Authors
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+    http://www.apache.org/licenses/LICENSE-2.0
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+import React from 'react';
+import { TextInput } from 'carbon-components-react';
+import Add from '@carbon/icons-react/lib/add--alt/24';
+import Remove from '@carbon/icons-react/lib/misuse/24';
+import './SecretsModal.scss';
+
+const Annotations = props => {
+  const {
+    annotations,
+    handleChange,
+    handleAdd,
+    handleRemove,
+    invalidFields,
+    disabled
+  } = props;
+
+  const annotationFields = [];
+  for (let i = 0; i < annotations.length; i += 2) {
+    annotationFields.push(
+      <div className="annotationRow" key={i}>
+        <TextInput
+          id={`annotation${i}`}
```

Comment:
I'm using data-testid for testing that way I don't have to use `documents.getElementByID`

Reply:
Can we use `byLabelText` or one of the other queries on some user-facing attributes?

Reply:
if I add a label text each input element will display that same label text which will distort the whole design... can't we stick with data-testid?

Reply:
Each input must have some sort of label / description (label, title, aria-label, aria-labelledby, etc.) for accessibility purposes, otherwise screen reader users would get a generic announcement similar to "input, type text to edit" and no indication what the field relates to. Whatever mechanism is used does not necessarily have to be visible on the screen, but may be useful to other users anyway.

Reply:
true, I'll add aria-labels & descriptions & see if I can test them with those instead of the testid